core/templates/dev/head/domain/classifier/PythonProgramTokenizerSpec.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests python program tokenizer.
+ */
+
+describe('Python program tokenizer', function() {
+  beforeEach(module('oppia'));
+
+  describe('Test python program tokenizer', function() {
+    var tokenizer;
+    beforeEach(inject(function($injector) {
+      tokenizer = $injector.get('PythonProgramTokenizer');
+    }));
+
+    it('should generate correct tokens for a program', function() {
+      var program = (
+        '# In Python, the code\n#\n#     for letter in [\'a\', \'b\']:\n#    ' +
+        '     print letter\n#\n# prints:\n#\n#     a\n#     b\ns = 0;\nfor ' +
+        'num in range(1000):\n  if num%7 == 0 or num%5 == 0:\n\ts +=x\n' +
+        'print s');
+
+      var expectedTokens = [
```

Comment:
Hmm, I'm not sure what impact this could have (the impact could be negligible or it might decrease the accuracy significantly). If we are splitting on '\n', then, can't we just add '\n' tokens later?